Tests/SonarQube.MSBuild.Tasks.IntegrationTests/TargetsTests/WriteProjectInfoFileTargetTests.cs

```diff
@@ -193,6 +200,64 @@ public void WriteProjectInfo_TestProject_HasTestGuid_LowerCase()
 
             // Assert
             AssertIsTestProject(projectInfo);
+            AssertProjectIsNotExcluded(projectInfo);
+        }
+
+        #endregion
+
+        #region Fakes projects tests
+
+        [TestMethod]
+        [TestCategory("ProjectInfo")] // SONARMSBRU-26: MS Fakes should be excluded from analysis
+        public void WriteProjectInfo_FakesProjectsAreExcluded()
+        {
+            // Checks that fakes projects are excluded from analysis
+
+            // Arrange
+            string rootInputFolder = TestUtils.CreateTestSpecificFolder(this.TestContext, "Inputs");
+            string rootOutputFolder = TestUtils.CreateTestSpecificFolder(this.TestContext, "Outputs");
+
+            EnsureAnalysisConfig(rootInputFolder, "pattern that won't match anything");
+
+            WellKnownProjectProperties preImportProperties = CreateDefaultAnalysisProperties(rootInputFolder, rootOutputFolder);
+            preImportProperties.AssemblyName = "f.fAKes";
```

Comment:
Low Priority: You could modify an existing test to use an assembly name that contains "fakes" but does not end in "fakes"